packages/console/src/pages/Profile/modals/BasicUserInfoUpdateModal/index.tsx

```diff
@@ -0,0 +1,96 @@
+import type { AdminConsoleKey } from '@logto/phrases';
+import { useEffect, useState } from 'react';
+import ReactModal from 'react-modal';
+
+import Button from '@/components/Button';
+import ModalLayout from '@/components/ModalLayout';
+import TextInput from '@/components/TextInput';
+import { adminTenantEndpoint, meApi } from '@/consts';
+import { useStaticApi } from '@/hooks/use-api';
+import * as modalStyles from '@/scss/modal.module.scss';
+
+export type BasicUserField = 'avatar' | 'username' | 'name';
+
+type Props = {
+  field?: BasicUserField;
+  value: string;
+  isOpen?: boolean;
+  onClose: () => void;
+};
+
+const BasicUserInfoUpdateModal = ({ field, value: defaultValue, isOpen, onClose }: Props) => {
```

Comment:
nit: maybe can use `originalValue` instead of `defaultValue` to avoid misunderstanding

Reply:
hmmm, ok. I'll change it to `initialValue` instead. Thanks!